src/rest/impl/contracts.service.ts

```diff
@@ -1245,6 +1309,261 @@ export default class ContractsServiceImpl extends ContractsService {
     };
   }
 
+  /**
+   * Creates or updates SAP contract according to given contract database model
+   *
+   * @param contractModel contract database model
+   * @param deliveryPlace delivery place database model
+   * @param itemGroup item group database model
+   *
+   * @returns promise of created or updated SAP contract
+   */
+  private createOrUpdateSapContract = async (
```

Comment:
Split into own create and update functions

Reply:
Creating and updating SAP contract has been split inside createPrUpdateSapContract method. It needs to handle both, since one of our contracts is just a part of SAP contract, so even if we _create_ one contract, it might just need to _update_ the SAP contract, and the logic to check whether SAP contract needs to be created or just updated is too long to be included straight into create and update contract methods. This pattern mimics the upsert method of most databases.